contracts/staking/DelegationManager.sol

```diff
@@ -79,23 +86,55 @@ contract DelegationManager is IDelegationManager, ERC721Full, Lockable {
   }
 
   // after unstaking wait for WITHDRAWAL_DELAY, in order to claim stake back
-  function unstakeClaim(uint256 delegatorId, bytes proof) public onlyDelegator(delegatorId) {
-    // TODO: complete me
+  function unstakeClaim(
+    uint256 checkpointId,// checkpoint Id  with root of proofs
+    uint256 delegatorId,
+    uint256 rewardAmount,
+    uint256 slashedAmount,
+    uint256 accIndex,
+    uint256 withdrawIndex,
+    bytes accProof,
+    bytes withdrawProof
+    ) public onlyDelegator(delegatorId) {
     StakeManager stakeManager = StakeManager(registry.getStakeManagerAddress());
-    // WITHDRAWAL_DELAY should match with validator's WITHDRAWAL_DELAY for perfect slashing
+    require(
+      keccak256(
+        abi.encodePacked(
+          delegatorId,
+          rewardAmount,
+          slashedAmount)
+          ).checkMembership(
+            accIndex,
+            accRoot[checkpointId],
+            accProof),
+      "Wrong account proof"
+      );
+    require(
+      keccak256(
+        abi.encodePacked(
+          delegatorId)
+          ).checkMembership(
+            withdrawIndex,
+            withdrawRoot[checkpointId],
+            withdrawProof),
+      "Wrong withdraw proof"
+      );
+
     require(
       delegators[delegatorId].deactivationEpoch > 0 &&
       delegators[delegatorId].deactivationEpoch.add(
       stakeManager.WITHDRAWAL_DELAY()) <= stakeManager.currentEpoch(),
-      ""
+      "Incomplete withdraw Period"
       );
+
     uint256 amount = delegators[delegatorId].amount;
     totalStaked = totalStaked.sub(amount);
```

Comment:
Also reduce the stake of validator in validatorDelegation

Reply:
this is handled in `_unbond`.